src/utils/internal/isGenerator.test.ts

```diff
@@ -0,0 +1,19 @@
+import { isGenerator } from './isGenerator'
+
+test('returns true given a generator function', () => {
+  expect(
+    isGenerator(function* () {
+      yield 2
+    }),
```

Comment:
This is currently failing because the function itself isn't a generator, but rather it produces a generator once called.

Reply:
An oversight, will provide the fix.